File: Models/Boiler.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace RajFabAPI.Models
{
    public class BoilerSpecifications
    {
        [Key]
        public Guid Id { get; set; }
        
        [Required]
        [MaxLength(50)]
        public string BoilerType { get; set; } = string.Empty; // fire-tube, water-tube, electric, waste-heat, other
        
        [Required]
        [MaxLength(100)]
        public string Manufacturer { get; set; } = string.Empty;
        
        [Required]
        [MaxLength(100)]
        public string SerialNumber { get; set; } = string.Empty;
        
        public int YearOfManufacture { get; set; }
        
        [Column(TypeName = "decimal(10,2)")]
        public decimal WorkingPressure { get; set; } // in kg/cm²
        
        [Column(TypeName = "decimal(10,2)")]
        public decimal DesignPressure { get; set; } // in kg/cm²
        
        [Column(TypeName = "decimal(10,2)")]
        public decimal SteamCapacity { get; set; } // in tonnes/hour
        
        [Required]
        [MaxLength(50)]
        public string FuelType { get; set; } = string.Empty; // coal, oil, gas, biomass, electric, multi-fuel
        
        [Column(TypeName = "decimal(10,2)")]
        public decimal HeatingArea { get; set; } // in m²
        
        [Column(TypeName = "decimal(10,2)")]
        public decimal? SuperheaterArea { get; set; } // in m²
        
        [Column(TypeName = "decimal(10,2)")]
        public decimal? EconomiserArea { get; set; } // in m²
        
        [Column(TypeName = "decimal(10,2)")]
        public decimal? AirPreheaterArea { get; set; } // in m²
        
        public DateTime CreatedAt { get; set; } = DateTime.Now;
        public DateTime UpdatedAt { get; set; } = DateTime.Now;
    }

    public class BoilerLocation
    {
        [Key]
        public Guid Id { get; set; }
        
        [Required]
        [MaxLength(200)]
        public string FactoryName { get; set; } = string.Empty;
        
        [MaxLength(100)]
        public string? FactoryLicenseNumber { get; set; }
        
        [Required]
        [MaxLength(100)]
        public string PlotNumber { get; set; } = string.Empty;
        
        [Required]
        [MaxLength(200)]
        public string Street { get; set; } = string.Empty;
        
        [Required]
        [MaxLength(100)]
        public string Locality { get; set; } = string.Empty;
        
        [Required]
        [MaxLength(10)]
        public string Pincode { get; set; } = string.Empty;
        
        public Guid AreaId { get; set; }
        public Guid CityId { get; set; }
        public Guid DistrictId { get; set; }
        public Guid DivisionId { get; set; }
        
        [Column(TypeName = "decimal(10,8)")]
        public decimal? Latitude { get; set; }
        
        [Column(TypeName = "decimal(11,8)")]
        public decimal? Longitude { get; set; }
        
        public DateTime CreatedAt { get; set; } = DateTime.Now;
        public DateTime UpdatedAt { get; set; } = DateTime.Now;
        
        // Navigation properties
        [ForeignKey("AreaId")]
        public Area? Area { get; set; }
        
        [ForeignKey("CityId")]
        public City? City { get; set; }
        
        [ForeignKey("DistrictId")]
        public District? District { get; set; }
        
        [ForeignKey("DivisionId")]
        public Division? Division { get; set; }
    }

    public class BoilerSafetyFeatures
    {
        [Key]
        public Guid Id { get; set; }
        
        [Required]
        [Column(TypeName = "nvarchar(max)")]
        public string SafetyValves { get; set; } = "[]"; // JSON array of safety valve objects
        
        public int WaterGauges { get; set; }
        public int PressureGauges { get; set; }
        public int? FusiblePlugs { get; set; }
        public int BlowdownValves { get; set; }
        
        [Required]
        [MaxLength(200)]
        public string FeedwaterSystem { get; set; } = string.Empty;
        
        public bool EmergencyShutoff { get; set; }
        
        public DateTime CreatedAt { get; set; } = DateTime.Now;
        public DateTime UpdatedAt { get; set; } = DateTime.Now;
    }

    public class BoilerCertificate
    {
        [Key]
        public Guid Id { get; set; }
        
        [Required]
        [MaxLength(100)]
        public string CertificateNumber { get; set; } = string.Empty;
        
        [Required]
        [MaxLength(50)]
        public string CertificateType { get; set; } = string.Empty; // initial, renewal, transfer, modification
        
        public DateTime IssueDate { get; set; }
        public DateTime ExpiryDate { get; set; }
        
        [Required]
        [MaxLength(200)]
        public string InspectorName { get; set; } = string.Empty;
        
        [Required]
        [MaxLength(100)]
        public string InspectorId { get; set; } = string.Empty;
        
        [Required]
        [MaxLength(50)]
        public string Status { get; set; } = "active"; // active, expired, suspended, cancelled
        
        public DateTime CreatedAt { get; set; } = DateTime.Now;
        public DateTime UpdatedAt { get; set; } = DateTime.Now;
    }

    public class BoilerInspectionHistory
    {
        [Key]
        public Guid Id { get; set; }
        
        [Required]
        [MaxLength(100)]
        public string InspectionId { get; set; } = string.Empty;
        
        public DateTime InspectionDate { get; set; }
        
        [Required]
        [MaxLength(50)]
        public string InspectionType { get; set; } = string.Empty; // annual, biennial, special, modification
        
        [Required]
        [MaxLength(200)]
        public string InspectorName { get; set; } = string.Empty;
        
        [Required]
        [Column(TypeName = "nvarchar(max)")]
        public string Findings { get; set; } = string.Empty;
        
        [Required]
        [Column(TypeName = "nvarchar(max)")]
        public string Recommendations { get; set; } = string.Empty;
        
        public DateTime NextInspectionDue { get; set; }
        public bool CertificateIssued { get; set; }
        
        public Guid BoilerId { get; set; }
        
        public DateTime CreatedAt { get; set; } = DateTime.Now;
        public DateTime UpdatedAt { get; set; } = DateTime.Now;
        
        // Navigation property
        [ForeignKey("BoilerId")]
        public RegisteredBoiler? Boiler { get; set; }
    }

    public class RegisteredBoiler
    {
        [Key]
        public Guid Id { get; set; }
        
        [Required]
        [MaxLength(100)]
        public string RegistrationNumber { get; set; } = string.Empty;
        
        public Guid SpecificationsId { get; set; }
        public Guid LocationId { get; set; }
        public Guid SafetyFeaturesId { get; set; }
        public Guid CurrentCertificateId { get; set; }
        
        [Required]
        [MaxLength(100)]
        public string OwnerId { get; set; } = string.Empty;
        
        [Required]
        [MaxLength(200)]
        public string OwnerName { get; set; } = string.Empty;
        
        [Required]
        [MaxLength(200)]
        public string OperatorName { get; set; } = string.Empty;
        
        [Required]
        [MaxLength(100)]
        public string OperatorCertificateNumber { get; set; } = string.Empty;
        
        public DateTime OperatorCertificateExpiry { get; set; }
        
        public DateTime RegistrationDate { get; set; }
        public DateTime LastModified { get; set; } = DateTime.Now;
        
        [Required]
        [MaxLength(50)]
        public string Status { get; set; } = "active"; // active, inactive, transferred, scrapped
        
        public DateTime CreatedAt { get; set; } = DateTime.Now;
        public DateTime UpdatedAt { get; set; } = DateTime.Now;
        
        // Navigation properties
        [ForeignKey("SpecificationsId")]
        public BoilerSpecifications? Specifications { get; set; }
        
        [ForeignKey("AreaId")]
        public BoilerLocation? Location { get; set; }
        
        [ForeignKey("SafetyFeaturesId")]
        public BoilerSafetyFeatures? SafetyFeatures { get; set; }
        
        [ForeignKey("CurrentCertificateId")]
        public BoilerCertificate? CurrentCertificate { get; set; }
        
        // Collection navigation properties
        public ICollection<BoilerInspectionHistory> InspectionHistory { get; set; } = new List<BoilerInspectionHistory>();
        public ICollection<BoilerApplication> Applications { get; set; } = new List<BoilerApplication>();
    }

    public class BoilerApplication
    {
        [Key]
        public Guid Id { get; set; }
        
        [Required]
        [MaxLength(100)]
        public string ApplicationNumber { get; set; } = string.Empty;
        
        [Required]
        [MaxLength(50)]
        public string ApplicationType { get; set; } = string.Empty; // registration, renewal, modification, transfer
        
        [Required]
        [MaxLength(50)]
        public string Status { get; set; } = "pending"; // pending, approved, rejected, under-review
        
        public Guid? BoilerId { get; set; }
        
        [Required]
        [MaxLength(200)]
        public string ApplicantName { get; set; } = string.Empty;
        
        [Required]
        [MaxLength(200)]
        public string OrganizationName { get; set; } = string.Empty;
        
        [Required]
        [MaxLength(100)]
        public string ContactPerson { get; set; } = string.Empty;
        
        [Required]
        [MaxLength(15)]
        public string Mobile { get; set; } = string.Empty;
        
        [Required]
        [MaxLength(200)]
        public string Email { get; set; } = string.Empty;
        
        [Required]
        [Column(TypeName = "nvarchar(max)")]
        public string Address { get; set; } = string.Empty;
        
        [Column(TypeName = "nvarchar(max)")]
        public string ApplicationData { get; set; } = "{}"; // JSON data for specific application type
        
        [Column(TypeName = "nvarchar(max)")]
        public string DocumentPaths { get; set; } = "[]"; // JSON array of document file paths
        
        public DateTime SubmissionDate { get; set; } = DateTime.Now;
        public DateTime? ProcessingDate { get; set; }
        public DateTime? CompletionDate { get; set; }
        
        [MaxLength(200)]
        public string? ProcessedBy { get; set; }
        
        [Column(TypeName = "nvarchar(max)")]
        public string? Comments { get; set; }
        
        public DateTime CreatedAt { get; set; } = DateTime.Now;
        public DateTime UpdatedAt { get; set; } = DateTime.Now;
        
        // Navigation property
        [ForeignKey("BoilerId")]
        public RegisteredBoiler? Boiler { get; set; }
    }
}

File: RajFABFrontend/src/types/boiler.ts
// Boiler service types based on Indian Boilers Act 1923 and IBR 1950

export interface BoilerSpecifications {
  boilerType: 'fire-tube' | 'water-tube' | 'electric' | 'waste-heat' | 'other';
  manufacturer: string;
  serialNumber: string;
  yearOfManufacture: number;
  workingPressure: number; // in kg/cm²
  designPressure: number; // in kg/cm²
  steamCapacity: number; // in tonnes/hour
  fuelType: 'coal' | 'oil' | 'gas' | 'biomass' | 'electric' | 'multi-fuel';
  heatingArea: number; // in m²
  superheaterArea?: number; // in m²
  economiserArea?: number; // in m²
  airPreheaterArea?: number; // in m²
}

export interface BoilerLocation {
  factoryName: string;
  factoryLicenseNumber?: string;
  plotNumber: string;
  street: string;
  locality: string;
  pincode: string;
  areaId: string;
  cityId: string;
  districtId: string;
  divisionId: string;
  latitude?: number;
  longitude?: number;
}

export interface BoilerSafetyFeatures {
  safetyValves: {
    count: number;
    settingPressure: number;
    capacity: number;
  }[];
  waterGauges: number;
  pressureGauges: number;
  fusiblePlugs?: number;
  blowdownValves: number;
  feedwaterSystem: string;
  emergencyShutoff: boolean;
}

export interface BoilerCertificate {
  certificateNumber: string;
  certificateType: 'initial' | 'renewal' | 'transfer' | 'modification';
  issueDate: string;
  expiryDate: string;
  inspectorName: string;
  inspectorId: string;
  status: 'active' | 'expired' | 'suspended' | 'cancelled';
}

export interface BoilerInspectionHistory {
  inspectionId: string;
  inspectionDate: string;
  inspectionType: 'annual' | 'biennial' | 'special' | 'modification';
  inspectorName: string;
  findings: string;
  recommendations: string;
  nextInspectionDue: string;
  certificateIssued: boolean;
}

export interface RegisteredBoiler {
  id: string;
  registrationNumber: string;
  specifications: BoilerSpecifications;
  location: BoilerLocation;
  safetyFeatures: BoilerSafetyFeatures;
  currentCertificate: BoilerCertificate;
  inspectionHistory: BoilerInspectionHistory[];
  ownerId: string;
  ownerName: string;
  operatorDetails: {
    name: string;
    certificateNumber: string;
    certificateExpiry: string;
  };
  registrationDate: string;
  lastModified: string;
  status: 'active' | 'inactive' | 'transferred' | 'scrapped';
}

// Form submission interfaces
export interface BoilerRegistrationForm {
  applicantInfo: {
    ownerName: string;
    organizationName: string;
    contactPerson: string;
    mobile: string;
    email: string;
    address: string;
  };
  specifications: BoilerSpecifications;
  location: BoilerLocation;
  safetyFeatures: BoilerSafetyFeatures;
  operatorDetails: {
    name: string;
    certificateNumber: string;
    certificateExpiry: string;
  };
  documents: {
    manufacturingCertificate: string;
    technicalDrawings: string;
    safetyValveCertificates: string;
    operatorCompetencyCertificate: string;
    locationApproval: string;
    environmentalClearance?: string;
  };
}

export interface BoilerRenewalForm {
  boilerRegistrationNumber: string;
  currentCertificateNumber: string;
  lastInspectionDate: string;
  renewalReason: string;
  changesFromLastInspection: string;
  operatorDetails: {
    name: string;
    certificateNumber: string;
    certificateExpiry: string;
  };
  documents: {
    lastInspectionReport: File[];
    currentBoilerPhotos: File[];
    operatorCompetencyCertificate: File[];
    maintenanceRecords: File[];
  };
}

export interface BoilerModificationForm {
  boilerRegistrationNumber: string;
  modificationType: 'pressure-increase' | 'capacity-increase' | 'fuel-change' | 'safety-upgrade' | 'location-change' | 'other';
  modificationDetails: string;
  engineeringJustification: string;
  proposedChanges: {
    currentSpecs: Partial<BoilerSpecifications>;
    proposedSpecs: Partial<BoilerSpecifications>;
  };
  safetyImpactAssessment: string;
  documents: {
    engineeringDrawings: File[];
    calculationSheets: File[];
    materialCertificates: File[];
    safetyAnalysis: File[];
  };
}

export interface BoilerTransferForm {
  boilerRegistrationNumber: string;
  transferType: 'ownership' | 'location' | 'both';
  currentOwner: {
    name: string;
    organizationName: string;
    contactInfo: string;
  };
  newOwner: {
    name: string;
    organizationName: string;
    contactPerson: string;
    mobile: string;
    email: string;
    address: string;
  };
  newLocation?: BoilerLocation;
  transferReason: string;
  operatorDetails: {
    name: string;
    certificateNumber: string;
    certificateExpiry: string;
  };
  documents: {
    ownershipTransferDeed: File[];
    locationApproval?: File[];
    operatorCompetencyCertificate: File[];
    currentCertificate: File[];
  };
}

// IBR Form types (Indian Boiler Regulations)
export const IBR_FORMS = {
  FORM_I: 'Registration/Inspection Book',
  FORM_II: 'Certificate of Inspection', 
  FORM_III: 'Application for Registration',
  FORM_IV: 'Application for Transfer',
  FORM_V: 'Modification Application',
} as const;

export type IBRFormType = keyof typeof IBR_FORMS;

// Boiler permission types
export const BOILER_PERMISSIONS = {
  VIEW_BOILER_RECORDS: { code: 'VIEW_BOILER_RECORDS', name: 'View Boiler Records' },
  REGISTER_BOILER: { code: 'REGISTER_BOILER', name: 'Register New Boiler' },
  RENEW_CERTIFICATE: { code: 'RENEW_CERTIFICATE', name: 'Renew Boiler Certificate' },
  MODIFY_BOILER: { code: 'MODIFY_BOILER', name: 'Approve Boiler Modifications' },
  TRANSFER_BOILER: { code: 'TRANSFER_BOILER', name: 'Process Boiler Transfers' },
  INSPECT_BOILER: { code: 'INSPECT_BOILER', name: 'Conduct Boiler Inspections' },
  ISSUE_CERTIFICATE: { code: 'ISSUE_CERTIFICATE', name: 'Issue Boiler Certificates' },
} as const;

export type BoilerPermissionCode = keyof typeof BOILER_PERMISSIONS;

// New payload types for boiler registration API
export interface OwnerDetailPayload {
  id?: string;
  name: string;
  designation: string;
  role: string;
  typeOfEmployer: string;
  relationType: string;
  relativeName: string;
  addressLine1: string;
  addressLine2: string;
  district: string;
  tehsil: string;
  area: string;
  pincode: string;
  email: string;
  telephone: string;
  mobile: string;
}

export interface MakerDetailPayload {
  id?: string;
  name: string;
  designation: string;
  role: string;
  typeOfEmployer: string;
  relationType: string;
  relativeName: string;
  addressLine1: string;
  addressLine2: string;
  district: string;
  tehsil: string;
  area: string;
  pincode: string;
  email: string;
  telephone: string;
  mobile: string;
}

export interface BoilerDetailPayload {
  addressLine1: string;
  addressLine2: string;
  districtId: string;
  subDivisionId: string;
  tehsilId: string;
  area: number;
  pinCode: number;
  renewalYears: number;
  telephone: string;
  mobile: string;
  email: string;
  erectionTypeId: number;
  makerNumber: string;
  yearOfMake: number;
  heatingSurfaceArea: number;
  evaporationCapacity: number;
  evaporationUnit: string;
  intendedWorkingPressure: number;
  pressureUnit: string;
  boilerTypeID: number;
  boilerCategoryID: number;
  superheater: boolean;
  superheaterOutletTemp: number;
  economiser: boolean;
  economiserOutletTemp: number;
  furnaceTypeID: number;
  drawingsPath: string;
  specificationPath: string;
  formI_B_CPath: string;
  formI_DPath: string;
  formI_EPath: string;
  formIV_APath: string;
  formV_APath: string;
  testCertificatesPath: string;
  weldRepairChartsPath: string;
  pipesCertificatesPath: string;
  tubesCertificatesPath: string;
  castingCertificatePath: string;
  forgingCertificatePath: string;
  headersCertificatePath: string;
  dishedEndsInspectionPath: string;
  boilerAttendantCertificatePath: string;
  boilerOperationEngineerCertificatePath: string;
}

export interface FactoryDetailsPayload {
  factoryName: string;
  factoryRegistrationNumber: string;
  addressLine1: string;
  addressLine2: string;
  districtId: string;
  subDivisionId: string;
  tehsilId: string;
  area: string;
  pinCode: string;
  mobile: string;
  telephone: string;
  email: string;
  erectionType: string;
}

export interface BoilerRegistrationPayload {
  factoryDetails: FactoryDetailsPayload;
  ownerDetail: OwnerDetailPayload;
  makerDetail: MakerDetailPayload;
  boilerDetail: BoilerDetailPayload;
}

// Boiler Manufacture Registration Types
export interface BMFacilityAddress {
  description: string;
  addressLine1: string;
  addressLine2: string;
  districtId: string;
  subDivisionId: string;
  tehsilId: string;
  area: number;
  pinCode: number;
}

export interface BMDesignFacility extends BMFacilityAddress {
  document: string;
}

export interface BMTestingFacility extends BMFacilityAddress {
  testingFacilityJson: string;
}

export interface BMRDFacility extends BMFacilityAddress {
  rdFacilityJson: string;
}

export interface BMNDTPersonnel {
  name: string;
  qualification: string;
  certificate: string;
}

export interface BMQualifiedWelder {
  name: string;
  qualification: string;
  certificate: string;
}

export interface BMTechnicalManpower {
  name: string;
  fatherName: string;
  qualification: string;
  minimumFiveYearsExperienceDoc: string;
  experienceInErectionDoc: string;
  experienceInCommissioningDoc: string;
}

// Create/Request payload for Boiler Manufacture Registration (without system-generated fields)
export interface BoilerManufactureCreatePayload {
  factoryRegistrationNo: string;
  bmClassification: string;
  coveredArea: string;
  establishmentJson: string;
  manufacturingFacilityjson: string;
  detailInternalQualityjson: string;
  otherReleventInformationjson: string;
  designFacility: BMDesignFacility;
  testingFacility: BMTestingFacility;
  rdFacility: BMRDFacility;
  ndtPersonnels: BMNDTPersonnel[];
  qualifiedWelders: BMQualifiedWelder[];
  technicalManpowers: BMTechnicalManpower[];
}

// Amend/Update payload for Boiler Manufacture Registration (includes applicationId for identification)
export interface BoilerManufactureAmendPayload {
  applicationId: string;
  factoryRegistrationNo: string;
  bmClassification: string;
  coveredArea: string;
  establishmentJson: string;
  manufacturingFacilityjson: string;
  detailInternalQualityjson: string;
  otherReleventInformationjson: string;
  designFacility: BMDesignFacility;
  testingFacility: BMTestingFacility;
  rdFacility: BMRDFacility;
  ndtPersonnels: BMNDTPersonnel[];
  qualifiedWelders: BMQualifiedWelder[];
  technicalManpowers: BMTechnicalManpower[];
}

// Alias for BoilerManufactureUpdatePayload used in API service
export type BoilerManufactureUpdatePayload = BoilerManufactureAmendPayload;

// Closure payload for Boiler Manufacture Registration
export interface BoilerManufactureClosurePayload {
  manufactureRegistrationNo: string;
  closureReason: string;
  closureDate: string;
  remarks: string;
  documentPath: string;
}

// Renewal response for Boiler Manufacture Registration
export interface BoilerManufactureRenewalPayload {
  manufactureRegistrationNo: string;
  renewalYears: number;
}

export interface BoilerManufactureRegistration {
  applicationId: string;
  manufactureRegistrationNo: string;
  factoryRegistrationNo: string;
  bmClassification: string;
  coveredArea: string;
  validFrom: string;
  validUpto: string;
  status: 'Approved' | 'Pending' | 'Rejected' | 'UnderReview';
  type: 'new' | 'renewal' | 'modification';
  version: number;
  establishmentJson: string;
  manufacturingFacilityjson: string;
  detailInternalQualityjson: string;
  otherReleventInformationjson: string;
  designFacility: BMDesignFacility;
  testingFacility: BMTestingFacility;
  rdFacility: BMRDFacility;
  ndtPersonnels: BMNDTPersonnel[];
  qualifiedWelders: BMQualifiedWelder[];
  technicalManpowers: BMTechnicalManpower[];
}

export type BoilerManufactureRegistrationList = BoilerManufactureRegistration[];

// Boiler Repairer Types
export interface BoilerRepairerEngineer {
  name: string;
  designation: string;
  qualification: string;
  experienceYears: number;
  documentPath: string;
}

export interface BoilerRepairerWelder {
  name: string;
  designation: string;
  experienceYears: number;
  certificatePath: string;
}

export interface BoilerRepairerCreatePayload {
  factoryRegistrationNo: string;
  brClassification: string;
  establishmentJson: string;
  jobsExecutedJson: string;
  documentEvidence: string;
  approvalHistoryJson: string;
  rejectedHistoryJson: string;
  toolsAvailable: boolean;
  simultaneousSites: number;
  acceptsRegulations: boolean;
  acceptsResponsibility: boolean;
  canSupplyMaterial: boolean;
  qualityControlType: string;
  qualityControlDetailsjson: string;
  engineers: BoilerRepairerEngineer[];
  welders: BoilerRepairerWelder[];
}

export interface BoilerRepairerAmendPayload {
  applicationId?: string;
  factoryRegistrationNo: string;
  brClassification: string;
  establishmentJson: string;
  jobsExecutedJson: string;
  documentEvidence: string;
  approvalHistoryJson: string;
  rejectedHistoryJson: string;
  toolsAvailable: boolean;
  simultaneousSites: number;
  acceptsRegulations: boolean;
  acceptsResponsibility: boolean;
  canSupplyMaterial: boolean;
  qualityControlType: string;
  qualityControlDetailsjson: string;
  engineers: BoilerRepairerEngineer[];
  welders: BoilerRepairerWelder[];
}

export interface BoilerRepairerClosurePayload {
  repairerRegistrationNo: string;
  closureReason: string;
  closureDate: string;
  remarks: string;
  documentPath: string;
}

export interface BoilerRepairerRenewalPayload {
  repairerRegistrationNo: string;
  renewalYears: number;
}

// Steam Pipeline Types
export interface SteamPipelineCreatePayload {
  boilerApplicationNo: string;
  proposedLayout: string;
  consentLetterProvided: string;
  steamPipeLineDrawingNo: string;
  boilerMakerRegistrationNo: string;
  erectorName: string;
  factoryRegistrationNumber: string;
  factorydetailjson: string;
  pipeLengthUpTo100mm: number;
  pipeLengthAbove100mm: number;
  noOfDeSuperHeaters: number;
  noOfSteamReceivers: number;
  noOfFeedHeaters: number;
  noOfSeparatelyFiredSuperHeaters: number;
  formII: string;
  formIII: string;
  formIIIA: string;
  formIIIB: string;
  formIV: string;
  formIVA: string;
  drawing: string;
  supportingDocuments: string;
}

export interface SteamPipelineAmendPayload {
  boilerApplicationNo: string;
  proposedLayout: string;
  consentLetterProvided: string;
  steamPipeLineDrawingNo: string;
  boilerMakerRegistrationNo: string;
  erectorName: string;
  factoryRegistrationNumber: string;
  factorydetailjson: string;
  pipeLengthUpTo100mm: number;
  pipeLengthAbove100mm: number;
  noOfDeSuperHeaters: number;
  noOfSteamReceivers: number;
  noOfFeedHeaters: number;
  noOfSeparatelyFiredSuperHeaters: number;
  formII: string;
  formIII: string;
  formIIIA: string;
  formIIIB: string;
  formIV: string;
  formIVA: string;
  drawing: string;
  supportingDocuments: string;
}

export interface SteamPipelineUpdatePayload {
  boilerApplicationNo: string;
  proposedLayout: string;
  consentLetterProvided: string;
  steamPipeLineDrawingNo: string;
  boilerMakerRegistrationNo: string;
  erectorName: string;
  factoryRegistrationNumber: string;
  factorydetailjson: string;
  pipeLengthUpTo100mm: number;
  pipeLengthAbove100mm: number;
  noOfDeSuperHeaters: number;
  noOfSteamReceivers: number;
  noOfFeedHeaters: number;
  noOfSeparatelyFiredSuperHeaters: number;
  formII: string;
  formIII: string;
  formIIIA: string;
  formIIIB: string;
  formIV: string;
  formIVA: string;
  drawing: string;
  supportingDocuments: string;
}

export interface SteamPipelineRenewPayload {
  steamPipeLineRegistrationNo: string;
  renewalYears: number;
  supportingDocumentsPath: string;
}

export interface SteamPipelineClosePayload {
  steamPipeLineRegistrationNo: string;
  reasonForClosure: string;
  supportingDocument: string;
}
